wasm-chain-integration/src/lib.rs

```diff
@@ -302,19 +318,30 @@ pub struct ReceiveHost<'a> {
     /// Outcomes of the execution, i.e., the actions tree.
     pub outcomes:          Outcome,
     /// The receive context for this call.
-    pub receive_ctx:       &'a ReceiveContext<&'a [u8]>,
+    pub receive_ctx:       &'a Ctx,
 }
 
-pub trait HasCommon {
+pub trait HasCommon<P, A>
+where
+    P: SerialPolicies<A>,
+    A: AsRef<[u8]>, {
```

Comment:
Is there a concrete reason for why A and P are parameters as opposed to associated types like MetadataType?

The usual reason for this is that you want different instances for the same type, but different parameters. Is that the case?

Please document if so.